python-avd/pyavd/_eos_designs/structured_config/base/__init__.py

```diff
@@ -48,7 +48,7 @@ def router_bgp(self) -> dict | None:
 
         router_bgp set based on switch.bgp_as, switch.bgp_defaults, router_id facts and aggregating the values of bgp_maximum_paths and bgp_ecmp variables.
         """
-        if self.shared_utils.bgp_as is None:
+        if self.shared_utils.bgp_as is None and not self.shared_utils.l3_interfaces_bgp_neighbors:
```

Comment:
The check should really be in the shared_utils `bgp` method like all the other checks.

Reply:
I am not sure I understand which method you are talking about - this code will stop if there is no no `bgp_as` - are you saying `bgp_as` should check on this?